### ISSUE TITLE
Conditionals supported?

### ISSUE THREAD

**Issue by alexbw** [JuliaDiff/ReverseDiffSource.jl#39]
I see you have support for some control flow (for-loops), which is awesome. I've been poking around the source, and don't see support for if-statements, while loops, or try/catch statements. Am I correct that these aren't in the library currently?

**Comment by fredo-dedup** [comment]
Yes, correct, only for-loops and provided they are not too complex in their structure (no dependencies between results of iterations).
I am working on a rewrite (devl3 branch)  that will support if-else statements, but I am only half-way through it and with very limited time to work on it.

**Comment by alexbw** [comment]
Hey, wanted to follow-up and see if you've made any progress? Interested in learning and discussing some of the ideas behind the package.
